open-hackathon-adminUI/src/app/views.py

```diff
@@ -48,16 +70,59 @@ def is_accessible(self):
         return True
 
 
-class AdminLoginView(AdminIndexView):
-    @expose('/')
-    def index(self):
-        return self.render_admin('login.html')
-
-
 admin = Admin(name="Open Hackathon Admin Console",base_template='admin/osslayout.html',index_view=HomeView())
 admin.init_app(app)
+
 admin.add_view(MyAdminView(name="view1", category='Test'))
 admin.add_view(AnotherAdminView(name="view2", category='Test'))
 
+###############################################################################################
+
+# js config
+@app.route('/config.js')
+def js_config():
+    resp = Response(response="var CONFIG=%s" % json.dumps(get_config("javascript")),
+                    status=200,
+                    mimetype="application/javascript")
+    return resp
+
+
+def __login(provider):
+    code = request.args.get('code')
+    login_result = login_providers[provider].login({
+        "code": code
+    })
+    admin = AdminUser(id=login_result["id"],
```

Comment:
Understand , make the session logic into background functions login() and logout()